cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -125,3 +164,194 @@ func return_{range_check_ptr, evm: Evm}() -> EthereumException* {
+        push{stack=stack}(zero);
+        EnvImpl.set_state{env=env}(state);
+        EvmImpl.set_env(env);
+        EvmImpl.set_stack(stack);
+        tempvar ok = cast(0, EthereumException*);
+        return ok;
+    }
+
+    increment_nonce{state=state}(evm.value.message.value.current_target);
+    EnvImpl.set_state{env=env}(state);
+    EvmImpl.set_env(env);
+
+    tempvar to = To(new ToStruct(new Bytes0(0), cast(0, Address*)));
+    // TODO: this could be optimized using a non-copy mechanism.
+    let (accessed_addresses_copy_start, accessed_addresses_copy) = dict_copy(
+        cast(evm.value.accessed_addresses.value.dict_ptr_start, DictAccess*),
+        cast(evm.value.accessed_addresses.value.dict_ptr, DictAccess*),
+    );
+    let (accessed_storage_keys_copy_start, accessed_storage_keys_copy) = dict_copy(
+        cast(evm.value.accessed_storage_keys.value.dict_ptr_start, DictAccess*),
+        cast(evm.value.accessed_storage_keys.value.dict_ptr, DictAccess*),
+    );
+    tempvar child_accessed_addresses = SetAddress(
+        new SetAddressStruct(
+            cast(accessed_addresses_copy_start, SetAddressDictAccess*),
+            cast(accessed_addresses_copy, SetAddressDictAccess*),
+        ),
+    );
+    tempvar child_accessed_storage_keys = SetTupleAddressBytes32(
+        new SetTupleAddressBytes32Struct(
+            cast(accessed_storage_keys_copy_start, SetTupleAddressBytes32DictAccess*),
+            cast(accessed_storage_keys_copy, SetTupleAddressBytes32DictAccess*),
+        ),
+    );
+    tempvar child_message = Message(
+        new MessageStruct(
+            caller=evm.value.message.value.current_target,
+            target=to,
+            current_target=contract_address,
+            gas=create_message_gas,
+            value=endowment,
+            data=empty_data_bytes,
+            code_address=cast(0, Address*),
+            code=call_data,
+            depth=Uint(evm.value.message.value.depth.value + 1),
+            should_transfer_value=bool(1),
+            is_static=bool(0),
+            accessed_addresses=child_accessed_addresses,
+            accessed_storage_keys=child_accessed_storage_keys,
+            parent_evm=evm,
+        ),
+    );
+
+    [ap] = range_check_ptr, ap++;
+    [ap] = bitwise_ptr, ap++;
+    [ap] = keccak_ptr, ap++;
+    [ap] = poseidon_ptr, ap++;
+    [ap] = child_message.value, ap++;
+    [ap] = env.value, ap++;
+
+    call abs process_create_message_label;
+
+    let range_check_ptr = [ap - 6];
+    let bitwise_ptr = cast([ap - 5], BitwiseBuiltin*);
+    let keccak_ptr = cast([ap - 4], KeccakBuiltin*);
+    let poseidon_ptr = cast([ap - 3], PoseidonBuiltin*);
+    let child_evm_ = cast([ap - 2], EvmStruct*);
+    let err = cast([ap - 1], EthereumException*);
+    tempvar child_evm = Evm(child_evm_);
+
+    // Note: we just returned `0` which, when casted into an EthereumException,
+    // is made into an object with value `0` instead of being a null ptr
```

Comment:
what is the difference?

Reply:
let ok = cast(0, EthereumException*);
    return ok;
```

Reply:
im not sure yet, this is just based on current observations that might still be bugged, this is not done